tests/lib/cli-engine.js

```diff
@@ -254,6 +254,21 @@ describe("CLIEngine", function() {
             });
         });
 
+        // https://github.com/eslint/eslint/issues/5547
+        it("should respect default ignore rules, even with --no-ignore", function() {
+
+            engine = new CLIEngine({
+                cwd: getFixturePath(),
+                ignore: false
+            });
+
+            var report = engine.executeOnText("var bar = foo;", "node_modules/passing.js");
+
+            assert.equal(report.results.length, 1);
+            assert.equal(report.results[0].filePath, getFixturePath("node_modules/passing.js"));
+            assert.equal(report.results[0].messages[0].message, "File ignored because of a matching ignore pattern. Use --no-ignore to override.");
```

Comment:
This warning proposes an invalid solution for default files, you are already using --no-ignore.